build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/plugin_list.rb

```diff
@@ -116,36 +150,27 @@ def signature_plugins(options)
         auth_types.each do |auth_type|
           case auth_type
           when 'v4'
-            plugins['Aws::Plugins::SignatureV4'] = "#{core_plugins}/signature_v4.rb"
+            plugins['Aws::Plugins::SignatureV4'] = "#{core_plugins_path}/signature_v4.rb"
           when 'v2'
-            plugins['Aws::Plugins::SignatureV2'] = "#{core_plugins}/signature_v2.rb"
+            plugins['Aws::Plugins::SignatureV2'] = "#{core_plugins_path}/signature_v2.rb"
           when 'bearer'
-            plugins['Aws::Plugins::BearerAuthorization'] = "#{core_plugins}/bearer_authorization.rb"
+            plugins['Aws::Plugins::BearerAuthorization'] = "#{core_plugins_path}/bearer_authorization.rb"
           end
         end
         plugins
       end
     end
 
-    def core_plugins
+    def core_plugins_path
       File.join(@aws_sdk_core_lib_path, 'aws-sdk-core/plugins')
     end
 
-    def seahorse_plugins
+    def seahorse_plugins_path
       File.join(@aws_sdk_core_lib_path, 'seahorse/client/plugins')
     end
 
-    def core_lib
-      # TODO : may need to register the default plugins directory rather
-      #        than have the hard-coded here as a relative path
-      File.expand_path('../../../../../gems/aws-sdk-core/lib', __FILE__)
-    end
-
     def const_get(class_name)
-      const_names = class_name.split('::')
-      const_names.inject(Kernel) do |const, const_name|
-        const.const_get(const_name)
-      end
+      Object.const_get(class_name)
```

Comment:
This is super old code from 9 years ago to get the const. I think the replacing line is fine? If so, I will get rid of the entire method entirely and just use `Object.const_get(...)` in the above code.